onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -324,7 +332,12 @@ - (Status)predict:(const std::unordered_map<std::string, OnnxTensorData>&)inputs
                                  ") do not match");
         }
 
-        const void* model_output_buffer = data.dataPointer;
+        ORT_RETURN_IF_NOT(IsArrayContiguous(data),
+                            "Non contiguous output MLMultiArray are not currently supported");
+        __block const void* model_output_buffer=nil;
+        [data getBytesWithHandler:^(const void *bytes, NSInteger size) {
+            model_output_buffer = bytes;
```

Comment:
should we also check that `size` is what we expect?

Reply:
Sure it won't hurt to be extra-sure and check for that too. Added it, it looks a bit ugly though as I have to check within the switch statement